epoch.go

```diff
@@ -172,7 +176,8 @@ func (e *Epoch) init() error {
 	e.sched = NewScheduler(e.Logger)
 	e.monitor = NewMonitor(e.StartTime, e.Logger)
 	e.cancelWaitForBlockNotarization = func() {}
-	e.finishCtx, e.finishFn = context.WithCancel(context.Background())
```

Comment:
can we revert this change? I know this sounds like Finnish context but `e.ctx` doesn't say what it's used for.